src/update-feeds-loops/update-feeds-loops.ts

```diff
@@ -241,6 +241,15 @@ export const runUpdateFeeds = async (providerName: ProviderName, chain: Chain, c
         dataFeedUpdates,
         dataFeedUpdateFailures,
       });
+
+      // Update the state with the signed API URLs.
+      const signedApiUrls = uniq(
+        processedBatches.reduce((acc, batch) => (batch ? [...acc, ...batch.signedApiUrls] : acc), [] as string[])
+      );
+      // Overwrite the state with the new signed API URLs instead of merging them to avoid stale URLs.
+      updateState((draft) => {
```

Comment:
I wanted to re-write `signedApiUrls` of `state` entirely instead of updating only changing parts. The rationale is when we phase out an airnode, old implementation doesn't remove this from the state and will continue to fetch and save signed data from signed API. I thought that most effective way to achieve this purpose is updating state after all batches processed. That's why I moved `updateState` statement in the func `processBatch` to here.

Reply:
> when we phase out an airnode

Not sure what this means.

Reply:
For example, when we want to remove one of the partner API provider (e.g. Nodary), we won't use its signed data so we don't need to make signed api call for its airnode address. If you see [here](https://github.com/api3dao/airseeker-v2/blob/a99a98acb3a8235edb15aadf8c1274e77281c1a1/src/update-feeds-loops/update-feeds-loops.ts#L279), previous implementation only creates or updates the existing records instead of re-creating draft.signedApiUrls entirely.

Reply:
Another example where this could be relevant is if the API provider rotates their mnemonic periodically for security purposes